java/src/main/java/com/logicalclocks/hsfs/FeatureGroup.java

```diff
@@ -226,6 +233,11 @@ public void insert(Dataset<Row> featureData, boolean overwrite, Map<String, Stri
     insert(featureData, null, overwrite, null, writeOptions);
   }
 
+  public void insert(Dataset<Row> featureData,  Map<String, String> writeOptions)
```

Comment:
Style change: Can you keep the order consistent and move this function up? Functions should be ordered from the "most generic" to the "most specific".